meta-mender-st-stm32mp/classes/stm32mp-gptimg.bbclass

```diff
@@ -0,0 +1,41 @@
+inherit image_types
+
+# It is not possible to specify this using MENDER_IMAGE_BOOTLOADER_FILE/OFFSET,
+# instead we use an append to "gptimg" to embed the bootloader
+
+# STM32MP requires the disk image to be GPT for it to find the bootloader
+
+IMAGE_CMD:gptimg:append:stcommon() {
+
+    #adding kernel and dts files to the boot partition
+    cp ${DEPLOY_DIR_IMAGE}/kernel/uImage ${WORKDIR}/bootfs/
+    cp ${DEPLOY_DIR_IMAGE}/kernel/${KERNEL_DEVICETREE} ${WORKDIR}/bootfs/
+
+    # Remove file generated by mender-part-images.bbclass
+    rm "$wks"
+
+    cat >> "$wks" <<EOF
+
+part metadata1 --source rawcopy --fsoptions "noauto" --part-name=metadata1 --sourceparams="file=${DEPLOY_DIR_IMAGE}/arm-trusted-firmware/metadata.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 512K --align 420
+part metadata2 --source rawcopy --fsoptions "noauto" --part-name=metadata2 --sourceparams="file=${DEPLOY_DIR_IMAGE}/arm-trusted-firmware/metadata.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 512K 
+part fip-a  --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fip-a --sourceparams="file=${DEPLOY_DIR_IMAGE}/fip/fip-stm32mp153a-ground4-mx-trusted.bin" --ondisk mmcblk --part-type 19d5df83-11b0-457b-be2c-7559c13142a5 --uuid="4FD84C93-54EF-463F-A7EF-AE25FF887087" --fixed-size 4096K
```

Comment:
these two lines are probably leftovers from a custom board, and fail the standard build

Reply:
I changed it to the right board. it should be in the gpt image.